cmd/storj/cmd/ls.go

```diff
@@ -32,49 +35,71 @@ func init() {
 func list(cmd *cobra.Command, args []string) error {
 	ctx := process.Ctx(cmd)
 
-	so, err := getStorjObjects(ctx, lsCfg)
+	identity, err := lsCfg.Load()
+	if err != nil {
+		return err
+	}
+
+	bs, err := lsCfg.GetBucketStore(ctx, identity)
 	if err != nil {
 		return err
 	}
 
 	if len(args) == 0 {
-		bi, err := so.ListBuckets(ctx)
-		if err != nil {
-			return err
+		startAfter := ""
+		var items []buckets.ListItem
+		for {
+			moreItems, more, err := bs.List(ctx, startAfter, "", 0)
+			if err != nil {
+				return err
+			}
+
+			items = append(items, moreItems...)
+			if !more {
+				break
+			}
+			startAfter = moreItems[len(moreItems)-1].Bucket
+		}
+
+		if len(items) == 0 {
```

Comment:
if you print the buckets out in the above for loop, the benefit is you wouldn't need to keep a potentially large list of buckets all in memory. the downside is you'd have to keep a flag on if you printed anything to determine if you should say "no buckets"

Reply:
Done